information/specification.md

```diff
@@ -0,0 +1,826 @@
+_roughly_ the same value **v**, then we will end up with a distribution of
+change values that matches the typical value distribution of payment
+requests.
+
+#### Principle 3: Performance Management
+
+Searching the UTxO set for additional entries to _improve_ our change outputs
+is _only_ useful if the UTxO set contains entries that are sufficiently
+small enough. But it is precisely when the UTxO set contains many small
+entries that it is less likely for a randomly-chosen UTxO entry to push the
+total above the upper bound.
+
+### Cardinality
+
+The Random-Improve algorithm imposes the following cardinality restriction:
+
+  * Each entry from the [initial UTxO set](#initial-utxo-set) is used to pay
+    for _at most one_ output from the [requested output
+    list](#requested-output-list).
+
+As a result of this restriction, the algorithm will fail with a [UTxO Not
+Fragmented Enough](#utxo-not-fragmented-enough) error if the number of entries
+in the [initial UTxO set](#initial-utxo-set) is _smaller than_ the number of
+entries in the [requested output list](#requested-output-list).
+
+### State
+
+At all stages of processing, the algorithm maintains the following pieces of
+state:
+
+ 1. #### Available UTxO Set
+
+    This is initially equal to the [initial UTxO set](#initial-utxo-set).
+
+ 2. #### Accumulated Coin Selection
+
+    The accumulated coin selection is a [coin selection](#coin-selection) where
+    all fields are initially equal to the _empty set_.
+
+### Computation
+
+The algorithm proceeds in two phases.
+
+#### Phase 1: Random Selection
+
+In this phase, the algorithm iterates through each of the [requested
+outputs](#requested-output-list) in descending order of coin value, from
+largest to smallest.
+
+For each output of value **v**, the algorithm repeatedly selects entries at
+**random** from the [available UTxO set](#available-utxo-set), until the _total
+value_ of selected entries is greater than or equal to **v**. The selected
+entries are then _associated with_ that output, and _removed_ from the
+[available UTxO set](#available-utxo-set).
+
+This phase ends when _every_ output has been associated with a selection of
+UTxO entries.
+
+#### Phase 2: Improvement
+
+In this phase, the algorithm attempts to improve upon each of the UTxO
+selections made in the previous phase, by conservatively expanding the
+selection made for each output in order to generate improved change
+values.
+
+During this phase, the algorithm:
+
+  * processes outputs in _ascending order of coin value_.
+
+  * continues to select values from the [available UTxO
+    set](#available-utxo-set).
+
+  * incrementally populates the
+    [accumulated coin selection](#accumulated-coin-selection-1).
+
+For each output of value **_v_**, the algorithm:
+
+ 1.  **Calculates a _target range_** for the total value of inputs used to
+     pay for that output, defined by the triplet:
+
+     (_minimum_, _ideal_, _maximum_) = (_v_, _2v_, _3v_)
```

Comment:
Okay so this is mentioned here. I think it'll deserve a small mention in the overview above, at least to explain the goal of the improvement step: getting change values closer to the output value.

Reply:
Done! See 4cf7b1334a40d382f2642ca1187a9f12a339f8d6.